File: 双周赛130A.py
class Solution:
    def satisfiesConditions(self, grid: List[List[int]]) -> bool:
        n, m = len(grid), len(grid[0])
        for i in range(n):
            for j in range(m):
                # 用整数非零时，bool 值为 True
                # i 就表示不在第一行，考虑前一行
                if i and grid[i][j] != grid[i-1][j]: return False
                # j 就表示不在第一列，考虑前一列
                if j and grid[i][j] == grid[i][j-1]: return False
        return True

File: 双周赛130A.cpp
class Solution {
public:
    bool satisfiesConditions(vector<vector<int>>& grid) {
        for (int i = 0; i < grid.size(); i++) {
            for (int j = 0; j < grid[0].size() - 1; j++) {
                if (grid[i][j] == grid[i][j + 1]) {
                    return false;
                }
            }
        }

        for (int i = 0; i < grid.size()-1; i++) {
            for (int j = 0; j < grid[0].size(); j++) {
                if (grid[i][j] != grid[i + 1][j]) {
                    return false;
                }
            }

        }
        return true;
    }
};
